internal/auth/api_server_mock.go

```diff
@@ -0,0 +1,62 @@
+package auth
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+type apiServerHandlerMock struct {
+	*http.ServeMux
+}
+
+func newAPIServerHandlerMock() *apiServerHandlerMock {
+	mux := http.NewServeMux()
+	s := &apiServerHandlerMock{ServeMux: mux}
+	mux.HandleFunc(APIAuthEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		if err := authHandler(w, r); err != nil {
+			log.Fatal(err)
+		}
+	})
+	return s
+}
+
+func authHandler(w http.ResponseWriter, r *http.Request) error {
```

Comment:
It does not make sense to return error from the handler as nothing can be done with it by `mux.HandleFunc`. So you can log errors directly here instead of wrapping `authHandler()` in another func.